src/rest/impl/deliveries.service.ts

```diff
@@ -604,105 +614,110 @@ export default class DeliveriesServiceImpl extends DeliveriesService {
           Quantity: loan.returned,
-          BinActionType: "batToWarehouse"
+          DocumentLinesBinAllocations: [
+            {
+              BinAbsEntry: 2,
+              Quantity: loan.returned,
+              BinActionType: BinActionTypeEnum.TO_WAREHOUSE,
+            },
+            {
+              BinAbsEntry: 3,
+              Quantity: loan.returned,
+              BinActionType: BinActionTypeEnum.FROM_WAREHOUSE
+            },
+          ]
         });
-        binAllocations.push({
-          BinAbsEntry: 3,
-          Quantity: loan.returned,
-          BinActionType: "batFromWarehouse"
-        });
-
-        const line: TransferLine = {
-          ItemCode: itemCode,
-          Quantity: loan.returned,
-          BinAllocations: binAllocations
-        };
-  
-        builder.addTransferLine(line);
       }
 
       if (loan.loaned > 0) {
-        const binAllocations: TransferLineBinAllocation[] = [];
-        binAllocations.push({
-          BinAbsEntry: 3,
+        stockTransferLines.push({
+          ...stockTransferLine,
           Quantity: loan.loaned,
-          BinActionType: "batToWarehouse"
+          DocumentLinesBinAllocations: [
+            {
+              BinAbsEntry: 3,
+              Quantity: loan.loaned,
+              BinActionType: BinActionTypeEnum.TO_WAREHOUSE
+            },
+            {
+              BinAbsEntry: 2,
+              Quantity: loan.loaned,
+              BinActionType: BinActionTypeEnum.FROM_WAREHOUSE
+            },
+          ]
         });
-        binAllocations.push({
-          BinAbsEntry: 2,
-          Quantity: loan.loaned,
-          BinActionType: "batFromWarehouse"
-        });
-
-        const line: TransferLine = {
-          ItemCode: itemCode,
-          Quantity: loan.loaned,
-          BinAllocations: binAllocations
-        };
-  
-        builder.addTransferLine(line);
       }
-
-
     });
 
-    const xml = builder.buildXML();
-    const filePath = `${config().sap["xml-fileupload-path"]}/${delivery.id ? delivery.id : Date.now().toString()}.xml`;
+    const stockTransfer: SapStockTransfer = {
+      DocDate: date,
+      CardCode: deliveryContactSapId,
+      Comments: notes,
+      SalesPersonCode: salesPersonCode,
+      FromWarehouse: loanWarehouse,
+      ToWarehouse: loanWarehouse,
+      StockTransferLines: stockTransferLines
+    };
 
-    return new Promise((resolve, reject) => {
-      fs.writeFile(filePath, xml, (err: any) => {
-        if (err) {
-          reject(err);
-        } else {
-          resolve();
-        }
-      });
-    });
+    try {
+      const sapPurchaseDeliveryNotesService = SapServiceFactory.getPurchaseDeliveryNotesService();
+      const sapStockTransfersService = SapServiceFactory.getStockTransfersService();
+      await sapPurchaseDeliveryNotesService.createPurchaseDeliveryNote(purchaseDeliveryNote);
+      await sapStockTransfersService.createStockTransfer(stockTransfer);
+    } catch (e) {
+      return Promise.reject(e);
```

Comment:
Log the error

Reply:
Here I was thinking that when this is called from createDelivery or updateDelivery as promises, they should be the ones to log the error and this should just return rejected promise with the error. Should the error log still be in here in this case?